assets/blocks/reader-registration/index.php

```diff
@@ -104,15 +104,18 @@ function render_block( $attrs, $content ) {
 		return '';
 	}
 
-	$registered       = false;
-	$my_account_url   = \wc_get_account_endpoint_url( 'dashboard' );
-	$message          = '';
-	$success_message  = __( 'Thank you for registering!', 'newspack' ) . '<br />';
-	$success_message .= sprintf(
-		// Translators: %s is a link to My Account.
-		__( 'Please visit %s to verify and manage your account.', 'newspack' ),
-		'<a href="' . esc_url( $my_account_url ) . '">' . __( 'My Account', 'newspack' ) . '</a>'
-	);
+	$registered      = false;
+	$my_account_url  = function_exists( 'wc_get_account_endpoint_url' ) ? \wc_get_account_endpoint_url( 'dashboard' ) : false;
+	$message         = '';
+	$success_message = __( 'Thank you for registering!', 'newspack' ) . '<br />';
+
+	if ( $my_account_url ) {
+		$success_message .= sprintf(
+			// Translators: %s is a link to My Account.
+			__( 'Please visit %s to verify and manage your account.', 'newspack' ),
+			'<a href="' . esc_url( $my_account_url ) . '">' . __( 'My Account', 'newspack' ) . '</a>'
```

Comment:
Is there ever a situation where a reader will see `Please visit  <a href="">My Account</a>` with no link in the case that `$my_account_url == false`?

Reply:
There shouldn't be—the link is only appended if `$my_account_url` is not empty.